Docs: hosting integrations for `build.commands`

humitos opened the pull request:
Initial file to start documenting hosting integrations.

This is not ready to review or merge. I'm just pushing this changes to ask for an early quick review to know if this is the direction we want to follow with this document. 

Document at https://docs--9755.org.readthedocs.build/en/9755/guides/hosting-integrations.html

Related #9036
Related https://github.com/readthedocs/meta/issues/71
Related https://github.com/readthedocs/meta/discussions/35

<!-- readthedocs-preview docs start -->
---
:books: Documentation previews :books:

- User's documentation (`docs`): https://docs--9755.org.readthedocs.build/en/9755/

<!-- readthedocs-preview docs end -->

<!-- readthedocs-preview dev start -->
- Developer's documentation (`dev`): https://dev--9755.org.readthedocs.build/en/9755/

<!-- readthedocs-preview dev end -->

ericholscher commented on docs/user/guides/hosting-integrations.rst:
```diff
@@ -0,0 +1,97 @@
+   Level up your search by providing immediate feedback while the user is typing.
+   Read more at :doc:`advanced-search.html#enable-search-as-you-type-in-your-documentation`.
+
+`Version warning`_
```
Hrm, I thought we had an extension for this?

humitos replied:
We have a core feature that handles the common use case _and_ a Sphinx extension that's more configurable. However, IIRC it broke some time ago due to CORS changes.

ericholscher commented on docs/user/guides/hosting-integrations.rst:
```diff
@@ -0,0 +1,97 @@
+   warning readers this is not the final version of this documentation.
+   Read more at :doc:`pull-requests`.
+
+.. TODO: write a new page explaining this and link it from here.
```
I'm pretty sure users won't care about integrating this, but the page explaining that we do it with some public metabase charts could be cool.

humitos replied:
Yeah, I'm not expecting final users to integrate this, but probably doctools/themes/etc. I think it's fine to have it documented and let people decide whether or not they want to support us by dumping this data. The more data they dump, the better support we can offer to them. So, it's a win-win in the end.

ericholscher reviewed (commented):
This looks like a great start. I think we can probably document this, but then have the goal to have the next step be to wrap all this up so you just need a single JS file that pulls the rest of that stuff in. 👍

benjaoming commented on docs/user/guides/hosting-integrations.rst:
```diff
@@ -0,0 +1,97 @@
+Hosting Integrations
+====================
+
+Read the Docs performs some actions by default when building documentation for its users to.
```
It's like something is off with this sentence?

But a good way to start it, could be like writing to the reader and the author and reviewers: "In this section, you can learn about <...>".

humitos replied:
I updated this a little to mention this as the first sentence. I think it's better now 👍🏼

benjaoming replied:
I still didn't entirely understand it, maybe it's just a redundant "to" at the end.

Here's what I mean by introducing the article in a way that tells both the author and the reader what the scope and intention of the article is... I think it's nice to start with, it can maybe be trimmed later.

```suggestion
Read the Docs performs a set of default actions to build and publish your documentation. In this article, we explain more about these actions, how they work, what they produce, how they relate. Ultimately we give you pointers on how to configure them.
```

benjaoming replied:
Great change :+1: 

I see also that "explain" is used, which is really good. At the end of content reorganizing, the idea of having a real reference for features will come up (so let's hope we are there in 1 month :crossed_fingers: ). In that regards, this overview and explanation will be valuable because it means that the feature reference can simply link to the explanation article.

benjaoming reviewed (commented):
I know this is a draft, so I want to just leave a few general inputs :)

* I think that this is _explanation_? In this regard, we should consider moving it out of the `guides/` folder because that folder is _mostly_ How-to. It would make a lot of sense if you already now suggest a folder for explanation, perhaps it's as easy as `explanation/`? (none of the current iterations have any proposals on where to put explanation)

* Start the section by writing _what it is about_. This might be shortened later, but it's really helpful in the beginning.

* Flyout menu: The word "Flyout" is super weird. I've put it in actual quotes in some page titles. Even what it refers to is confusing to me :D

>  A flyout occurs when a batter hits the ball in the air (not including balls designated as line drives) and an opposing defender catches it before it hits the ground or fence. 

 :question:  :confused: 

If you happen to think of something, please say so.. already in Iteration 1 https://github.com/readthedocs/readthedocs.org/issues/9746 we are looking at the pages about "Flyout" menus.

You've started explaining the "Flyout" menu already in a subsection, so that might also be a reason to coordinate things closely. Otherwise, this work might be easier in Iteration 3 or 4 when we have wrestled with the existing contents and have started to focus more on developing new content.

humitos commented:
@ericholscher I updated this page a little more adding more explanations for some of the other features. I added some notes to consider/discuss/talk about because telling people to "include this chunk of HTML" won't work for some features (e.g. version warning).

I think it is still fine to communicate "what we do", but I don't think it works as "integrate these features in your doctool" yet. IMO, we are not there yet and we should decouple them from being Sphinx extensions and migrate them into a Javascript library that does all this work (for the version warning example, adds the HTML, put the CSS to make it sticky at the top, generate the correct links, etc).

Writing this document is being a good exercise to realize what is required to make it fully compatible and generic enough, 💯

benjaoming commented on docs/user/guides/hosting-integrations.rst:
```diff
@@ -0,0 +1,211 @@
+Flyout menu
+-----------
+
+Displying the flyout menu requires adding some Javascript and CSS files in each of the documentation pages.
```
```suggestion
Displaying the flyout menu requires adding some JavaScript and CSS files in each of the documentation pages.
```

ericholscher commented:
@benjaoming Flyout is a standard term for what we're doing.. https://elementor.com/resources/glossary/what-is-flyout-navigation/

benjaoming commented:
Good reference to know. It'd still be interesting to figure out a better name. For now, I think we should resort to be careful to always mention it with a definition or reference.

To me as a reader, the word "flyout" is kind of alienating if it doesn't come with a definition or reference.

ericholscher commented:
@benjaoming Yea, I think in general we try to `:term:` it.

humitos commented:
📓 Note to myself:

> If you require more detailed analytics, Read the Docs has native support for Google Analytics.

(from https://github.com/readthedocs/readthedocs.org/pull/9677/files#diff-98cc398741c4326533ef8f26b360e24f914d4ef3998e69474d10a163e42eaa4dR87)

We need to explain here how this works and _where_ this is injected? How do we plan this to work on non-Sphinx projects?

humitos commented:
This is another potentially good integration feature: https://github.com/readthedocs/readthedocs.org/issues/9854

humitos commented:
I'm closing this PR for now since is superseded by #10127. The important conversation will be happening there.
